dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java

```diff
@@ -1576,6 +1577,134 @@ public final Response fireActionDefault(@Context final HttpServletRequest reques
+                                                 final InitDataObject initDataObject,
+                                                 final OutputStream outputStream,
+                                                 final ObjectMapper objectMapper) {
+
+        final DotSubmitter dotSubmitter = DotConcurrentFactory.getInstance().getSubmitter("workflow_submitter",
+                new DotConcurrentFactory.SubmitterConfigBuilder().poolSize(2).maxPoolSize(5).queueCapacity(CONTENTLETS_LIMIT).build());
+        final CompletionService<Map<String, Object>> completionService = new ExecutorCompletionService<>(dotSubmitter);
+        final List<Future<Map<String, Object>>> futures = new ArrayList<>();
+
+        for (final Map<String, Object> contentMap : contentletsToSaveList) {
+
+            // this triggers the save
+            final Future<Map<String, Object>> future = completionService.submit(() -> {
+
+                final Map<String, Object> resultMap = new HashMap<>();
+                final String inode      = (String) contentMap.get("inode");
+                final String identifier = (String) contentMap.get("identifier");
+                final long languageId   = ConversionUtils.toLong(contentMap.get("languageId"), -1l);
+                final User user         = initDataObject.getUser();
+                final FireActionForm singleFireActionForm = new FireActionForm.Builder()
+                    .contentlet(contentMap).assign(fireActionForm.getAssign())
+                    .comments(fireActionForm.getComments()).expireDate(fireActionForm.getExpireDate())
+                    .neverExpire(fireActionForm.getNeverExpire()).filterKey(fireActionForm.getFilterKey())
+                    .iWantTo(fireActionForm.getIWantTo()).publishDate(fireActionForm.getPublishDate())
+                    .publishTime(fireActionForm.getPublishTime()).whereToSend(fireActionForm.getWhereToSend()).build();
+                final IndexPolicy indexPolicy = MapToContentletPopulator.recoverIndexPolicy(
+                        singleFireActionForm.getContentletFormData(),
+                        IndexPolicyProvider.getInstance().forSingleContent(), request);
+                final Contentlet contentlet = this.getContentlet
+                        (inode, identifier, languageId,
+                                ()->WebAPILocator.getLanguageWebAPI().getLanguage(request).getId(),
+                                singleFireActionForm, initDataObject, mode);
+
+                try {
+
+                    fireTransactionalAction(systemAction, singleFireActionForm, request, mode,
+                            initDataObject, resultMap, inode, identifier, languageId, user, indexPolicy);
+                } catch (Exception e) {
+
+                    final String id = UtilMethods.isSet(identifier)?identifier:inode;
+                    Logger.error(this, "Error in contentlet: " + id + ", msg: " + e.getMessage(), e);
```

Comment:
Is it possible to include what specific action failed for this content? If it is, we need to log it.